File: Day2/day_2.pl
#!/usr/bin/perl
##
# advent of code 2022 - Day 2 (parts 1 & 2) [perl]
##
#
# Author: Eugene Dupler

$INFILE = "Day2/input_data.txt";

open(INFILE, "<$INFILE") || die("Could not open $INFILE");
@data = <INFILE>;
close(INFILE);

$s1 = $s2 = 0;

# Detect patterns and add values
foreach (@data) {
    if (/A X/) { $s1+=4 ; $s2+=3 };
    if (/A Y/) { $s1+=8 ; $s2+=4 };
    if (/A Z/) { $s1+=3 ; $s2+=8 };
    if (/B X/) { $s1+=1 ; $s2+=1 };
    if (/B Y/) { $s1+=5 ; $s2+=5 };
    if (/B Z/) { $s1+=9 ; $s2+=9 };
    if (/C X/) { $s1+=7 ; $s2+=2 };
    if (/C Y/) { $s1+=2 ; $s2+=6 };
    if (/C Z/) { $s1+=6 ; $s2+=7 };
}

print("score1 " .$s1."\n");
print("score2 ".$s2."\n");

File: Day2/day_2.py
"""
advent of code 2022 - Day 2 (parts 1 & 2) [python3]

Author: Eugene Dupler
"""

def main():
    # Open our data file
    input_file = "Day2/input_data.txt"


    points1 = {
        'A': { 'X':3, 'Y':6, 'Z':0 },
        'B': { 'X':0, 'Y':3, 'Z':6 },
        'C': { 'X':6, 'Y':0, 'Z':3 },
        'X': 1,
        'Y': 2,
        'Z': 3
    }

    points2 = {
        'A': { 'X':3, 'Y':1, 'Z':2 },
        'B': { 'X':1, 'Y':2, 'Z':3 },
        'C': { 'X':2, 'Y':3, 'Z':1 },
        'X': 0,
        'Y': 3,
        'Z': 6
    }

    try:
        with open(input_file, 'r') as file:
            data = file.read().splitlines()
    except Exception as e:
        print(e)

    score1 = score2 = 0
    for line in data:
        (p1, p2) = line.split()
        score1 += points1[p1][p2] + points1[p2]
        score2 += points2[p1][p2] + points2[p2]

    print("score1", score1)
    print("score2", score2)
    

if __name__ == "__main__":
    main()
